kolibri/auth/management/commands/fullfacilitysync.py

```diff
@@ -0,0 +1,74 @@
+from django.utils.six.moves import input
+from kolibri.auth.models import FacilityUser
+from kolibri.core.device.utils import device_provisioned
+from kolibri.core.device.models import DevicePermissions, DeviceSettings
+from kolibri.tasks.management.commands.base import AsyncCommand
+from morango.controller import MorangoProfileController
+from morango.certificates import Certificate, Filter
+
+class Command(AsyncCommand):
+
+    def add_arguments(self, parser):
+        parser.add_argument('--dataset-id', type=str)
+        parser.add_argument('--no-push', type=bool, default=False)
+        parser.add_argument('--no-pull', type=bool, default=False)
+        parser.add_argument('--host', type=str)
+        parser.add_argument('--username', type=str)
+        parser.add_argument('--password', type=str)
+
+    def handle_async(self, *args, **options):
+        controller = MorangoProfileController('facilitydata')
+        with self.start_progress(total=5) as progress_update:
+            network_connection = controller.create_network_connection(options['host'])
+            progress_update(1)
+            # get list of facilities and if more than 1, display all choices to user
```

Comment:
Yup, the use case stands firm! I think it's the original reason we need to get hold of, and if no one can remember it, we might have to introduce an "invisible to anon users" flag per facility for those particular cases?

Reply:
yeah well I think the original reason was when learners are signing up through kolibri and if there's multiple facilities only show the ones that allow learner sign ups per say. But I don't see the usefulness in that. There has to be a specific facility that a user should sign up to (they shouldn't be just signing up to random facilities). Unless we do introduce a flag like you said that just closes off a facility from the outside world and you only have access through logging in. Also, unless that was the real intention with the `learners_can_sign_up` flag, in that case maybe just rename the flag.

Reply:
The sensitive thing in this case seems to be the sign-up itself, but not about read access to some facility data -- this could be sensitive too though, but likely to be guarded better by normal login authentication.